apps/ejabberd/src/ejabberd_listener.erl

```diff
@@ -190,16 +193,24 @@ listen_tcp(PortIPProto, Module, SockOpts, Port, IPS) ->
             end
     end.
 
+override_sock_opts([], Opts) ->
+    Opts;
+override_sock_opts([Override | OverrideOpts], Opts) ->
+    NewOpts = do_override(Override, Opts),
+    override_sock_opts(OverrideOpts, NewOpts).
+
+do_override({ip, _} = IP, Opts) ->
+    lists:keystore(ip, 1, Opts, IP);
+do_override({backlog, _} = Backlog, Opts) ->
+    lists:keystore(backlog, 1, Opts, Backlog);
+do_override(inet6, Opts) ->
+    [inet6 | lists:delete(inet6, Opts)];
+do_override(inet, Opts) ->
+    [inet | lists:delete(inet, Opts)];
```

Comment:
No, it's not the same. What I wanted to achieve is to have only one `inet6` or `inet` value in the list.